internal/dot/graph.go

```diff
@@ -28,26 +50,34 @@ type Node struct {
 	Group    string
 }
 
-// Edge connects a node parameter to a node result.
-type Edge struct {
-	Param  *Node
-	Result *Node
+// Add updates the param and result nodes in the constructor and adds the constructor to the graph
+func (dg *Graph) Add(c *Ctor, params []*Node, results []*Node) {
+	c.Params = getNodes(dg, params)
+	c.Results = getNodes(dg, results)
+	dg.Ctors = append(dg.Ctors, c)
 }
 
-// Graph is the DOT-format graph in a Container.
-type Graph struct {
-	Edges []*Edge
+func (n *Node) str() string {
```

Comment:
Seeing as this is already in an `internal/` package, we can export this and implement the [`fmt.Stringer`](https://golang.org/pkg/fmt/#Stringer) interface:
```
func (dg *Graph) String() string { ... }
```

In general I agree with your inclination to un-export methods that aren't used outside of this package (as in `getNodes` below), but conforming to the `fmt.Stringer` interface is more appropriate here.